### PR TITLE
Remove logging validation

### PR DESCRIPTION
**Pull request by stanlyDoge**
Signed-off-by: Stanislav Knot <sknot@redhat.com>

### Type of change
- Enhancement

### Description
Fixes https://github.com/strimzi/strimzi-kafka-operator/issues/1848

### Checklist

- [ ] Update/write design documentation in `./design`
- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java [hunk 1]
```diff
@@ -341,57 +334,10 @@ protected void setLogging(Logging logging) {
...
-                    return;
-                }
-                if (key.contains("log4j.appender.CONSOLE")) {
-                    log.warn("You cannot set appender");
-                    return;
-                }
-                if ((asList(validLoggerValues).contains(tmpEntry.replaceAll(",[ ]+CONSOLE", ""))) || (asList(validLoggerValues).contains(tmpEntry))) {
-                    // correct value
-                } else {
-                    Matcher m = LOGGER_PATTERN.matcher(tmpEntry);
```
Comment (tombentley) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java]: I guess you can remove `LOGGER_PATTERN`.

**Comment by stanlyDoge** [comment]
Isn't this covered in https://strimzi.io/docs/master/#ref-kafka-logging-str ?

**Comment by tombentley** [comment]
Why do we even have that as an appendix? It's completely illogical to have it there when we have logging sections in each of the per-CR documentation (such as https://strimzi.io/docs/master/#assembly-logging-deployment-configuration-kafka-connect-s2i) which is the documentation people will actually be reading.

**Comment by stanlyDoge** [comment]
@PaulRMellor @laidan6000 ^

**Comment by tombentley** [comment]
I'd like to see the appendix removed, because I think it's weird to have this little appendix at the end when the bulk of the info about logging is covered in the per-CR documentation. But I disagree that the per-CR documentation is entirely adequate because:

* it does not make clear which components use log4j and which use log4j2 (in all cases they link to the log4j2 manual, which is wrong for Kafka and KafkaConnect, I think). 
* The inline logging example using _logger.name_ is confusing for users (i.e. we've actually seen people getting confused by this) since it's not adequately explained that it is a placeholder for a logger name. I think the example would be much better using an actual logger name that's used by that component (i.e. a Kafka logger name for the `Kafka` CR example, an actual UO logger name for the `KafkaUser` CR example, etc). 
* The external logging example talks about putting the whole logging config under the log4j.properties or log4j2.properties key in the config map but again doesn't say which to use. And again it could really use an example of what the config map should look like in each case, using an actual logger name that's used by that component.

**Comment by ppatierno** [comment]
I agree with you about example but you know that using actual loggers name per component doesn't help with modularized and reusable doc ;-)

**Comment by tombentley** [comment]
Why should we prioritise modular doc over documentation which actually helps people understand the software?

**Comment by ppatierno** [comment]
I don't know if you UK you have this but we said "you are pushing an opened door with me on this" ;)
So I totally agree with you.

**Comment by PaulRMellor** [comment]
@ppatierno @stanlyDoge @tombentley @laidan6000 

I think the appendix can be removed. The reason it is there is that it didn't naturally sit anywhere else. 

If we're going the way of Mirror Maker, we'll be reducing all the extraneous content down and using actual logger names in the examples. So we can have something like this for each: https://strimzi.io/docs/master/#con-kafka-mirror-maker-logging-deployment-configuration-kafka-mirror-maker.

I can create an issue to tidy all this up?

**Comment by tombentley** [comment]
Thanks @PaulRMellor, yes please create an issue.

**Comment by scholzj** [comment]
What is the state of this? Should we merge this?

**Comment by stanlyDoge** [comment]
Unless @tombentley has another opinion, I think yes.

**Comment by tombentley** [comment]
Yes, my only problem was around the documentation, whch @PaulRMellor seems to have taken on.

**Pull request merged**